hook.php

```diff
@@ -531,27 +496,16 @@ function plugin_formcreator_hook_purge_ticket(CommonDBTM $item) {
 
    $id = $item->getID();
 
-   // Update status of form answer, if any
+   // Update the formanswer's status (for cases where a form answer has several tickets)
    $formAnswer = new PluginFormcreatorFormAnswer();
-   $formAnswer->getFromDBByCrit([
-      'id' => new QuerySubQuery([
-         'SELECT' => 'items_id',
-         'FROM'   => Item_Ticket::getTable(),
-         'WHERE'  => [
-            'itemtype' => PluginFormcreatorFormAnswer::getType(),
-            'tickets_id' => $id,
-         ]
-      ])
-   ]);
-   if (!$formAnswer->isNewItem()) {
+   if ($formAnswer->getFromDbByTicket($id)) {
       $minimalStatus = $formAnswer->getAggregatedStatus();
       if ($minimalStatus === null) {
          // There is no more ticket in the form anwer
          $formAnswer->updateStatus(CommonITILObject::CLOSED);
       } else {
          $formAnswer->updateStatus($minimalStatus);
       }
-      return;
```

Comment:
after updatiing the form answer, we must delete the issue which represents tie ticket, if any (this happens when there is only 1 ticket associated to the form answer)